coderd/coderdtest/coderdtest.go

```diff
@@ -925,10 +930,19 @@ func MustTransitionWorkspace(t testing.TB, client *codersdk.Client, workspaceID
 	template, err := client.Template(ctx, workspace.TemplateID)
 	require.NoError(t, err, "fetch workspace template")
 
-	build, err := client.CreateWorkspaceBuild(ctx, workspace.ID, codersdk.CreateWorkspaceBuildRequest{
+	req := codersdk.CreateWorkspaceBuildRequest{
+		// TODO (JonA): I get this is for convenience but we should probably
+		// change this. Tripped me up why my test was passing when it shouldn't
+		// have.
```

Comment:
I don't really understand this comment --- change how?

Reply:
Currently the behavior when calling `MustTransitionWorkspace` is to submit a workspace build with the template's active version as opposed to the the workspaces template version. That's kinda at odds with how it behaves in production.